[GSOC 2025] : Action only Pull Request

Divyansh013 opened the pull request:
In continuation of #895

limzykenneth commented:
@two-ticks Looks good so far. Are you thinking about adding KaTeX equation formatting at this point or is this ready to merged?

two-ticks commented:
It is producing same result as processing.org so for now we can consider it to be ready for merge. 
 
<img src="https://user-images.githubusercontent.com/68433541/110792686-b4365980-8299-11eb-8f2a-74527cc0d57b.png"   height="400" />

two-ticks commented:
We can open separate issue for KaTeX equation formatting because other examples such as `Sine Cosine` also require KaTeX equation formatting.

<img src="https://user-images.githubusercontent.com/68433541/110792924-04adb700-829a-11eb-97bf-61b2f8f0d3ed.png"   height="400" />

lmccart commented on src/data/examples/en/08_Math/20_Graphing2DEquations.js:
```diff
@@ -0,0 +1,54 @@
+
+function draw() {
+  loadPixels();
+  var n = (mouseX * 10.0) / width;
```
can you update this to ES6 and use `const` or `let` for these?

two-ticks replied:
Changes : 

  let n = (mouseX * 10.0) / width;
  const w = 16.0; // 2D space width
  const h = 16.0; // 2D space height
  const dx = w / width; // Increment x this amount per pixel
  const dy = h / height; // Increment y this amount per pixel
  let x = -w / 2; // Start x at -1 * width / 2
  let y;

  let r;
  let theta;
  let val;

lmccart reviewed (changes requested):
can you please update to ES6 in all three files? thanks!

two-ticks commented:
updated es6 in all three files

lmccart commented:
looks great, thanks!

Divyansh013 committed:
Test: Add documentation line to circle.mdx for translation tracker testing

Divyansh013 committed:
Add Week 1 translation tracker implementation

Divyansh013 committed:
Test: Modify triangle.mdx for translation testing

Divyansh013 committed:
Test: Modify accelerationX.mdx to test missing translation detection

Divyansh013 committed:
removed testing logs

Divyansh013 committed:
small fix

Divyansh013 committed:
post review changes

Divyansh013 committed:
Week 2: Translation tracker with GitHub API and Hindi focus

Divyansh013 committed:
Test: Update English example for translation tracker testing

Divyansh013 committed:
Fix: Add automatic branch detection to translation tracker

Divyansh013 committed:
week 2 complete

Divyansh013 committed:
Week 3: Test automated translation tracking

- Updated translation-sync workflow for examples instead of reference
- Added Week 3 test change to Shape Primitives description
- Should trigger automated issue creation for outdated translations

Divyansh013 committed:
Add manual trigger support to translation tracker workflow

- Added workflow_dispatch with scan_all and create_issues options
- Can now manually trigger action from GitHub UI
- Supports both normal and full-scan modes

Divyansh013 committed:
Test: Manual trigger detection test

- Modified description.mdx to test translation tracker
- Should now detect 1 English example file change

Divyansh013 committed:
Simplify translation tracker - clean output

- test-local.js: Only shows changed files from predefined test files
- index.js: Production mode with minimal console output
- Removed verbose logging and repository exploration
- Clear distinction: test mode vs production mode with GitHub token

Divyansh013 committed:
Add scan all files option to index.js

- SCAN_ALL=true environment variable to check all 61 example files
- Simplified GitHub Actions workflow
- Default: checks only changed files
- Manual trigger: option to scan all files

Divyansh013 committed:
final fix

Divyansh013 committed:
temp commit

Divyansh013 committed:
basic version complete

Divyansh013 committed:
readme updated and simplified scanning

Divyansh013 committed:
review fix

Divyansh013 committed:
Add translation tracking GitHub Action

- Automated translation status tracking for p5.js examples
- Detects outdated/missing translations using Git commit comparison
- Creates consolidated GitHub issues with diff snippets
- Configured to monitor staging branch for translation changes
- Supports Spanish, Hindi, Korean, and Chinese Simplified
- Action-only implementation for iterative deployment (Step 2/4)

Divyansh013 commented:
@ksen0 have removed ui part from this PR, let me know if this is the correct way to do so

ksen0 committed:
Merge branch 'main' into action-only-main
